From dd4f19e7fa9edfb906cf7696335144851e237f3b Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Sun, 25 Oct 2020 21:50:30 +0100
Subject: [PATCH 104/107] HACK: drm/vmwgfx: pretend drm format ABGR/XBGR
 support (v3)

For testing and study purposes

(v2) AGBR/XBGR formats in vmw_create_bo_proxy()

(v3) AGBR/XBGR formats in vmw_kms_new_framebuffer_bo()
---
 drivers/gpu/drm/vmwgfx/vmwgfx_kms.c | 6 ++++++
 drivers/gpu/drm/vmwgfx/vmwgfx_kms.h | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/vmwgfx/vmwgfx_kms.c b/drivers/gpu/drm/vmwgfx/vmwgfx_kms.c
index 18b4add02cbc..1314b128a761 100644
--- a/drivers/gpu/drm/vmwgfx/vmwgfx_kms.c
+++ b/drivers/gpu/drm/vmwgfx/vmwgfx_kms.c
@@ -918,9 +918,11 @@ static int vmw_kms_new_framebuffer_surface(struct vmw_private *dev_priv,
 	}
 
 	switch (mode_cmd->pixel_format) {
+	case DRM_FORMAT_ABGR8888:
 	case DRM_FORMAT_ARGB8888:
 		format = SVGA3D_A8R8G8B8;
 		break;
+	case DRM_FORMAT_XBGR8888:
 	case DRM_FORMAT_XRGB8888:
 		format = SVGA3D_X8R8G8B8;
 		break;
@@ -1153,6 +1155,8 @@ static int vmw_create_bo_proxy(struct drm_device *dev,
 	switch (mode_cmd->pixel_format) {
 	case DRM_FORMAT_ARGB8888:
 	case DRM_FORMAT_XRGB8888:
+	case DRM_FORMAT_ABGR8888:
+	case DRM_FORMAT_XBGR8888:
 		format = SVGA3D_X8R8G8B8;
 		bytes_pp = 4;
 		break;
@@ -1229,6 +1233,8 @@ static int vmw_kms_new_framebuffer_bo(struct vmw_private *dev_priv,
 		switch (mode_cmd->pixel_format) {
 		case DRM_FORMAT_XRGB8888:
 		case DRM_FORMAT_ARGB8888:
+		case DRM_FORMAT_XBGR8888:
+		case DRM_FORMAT_ABGR8888:
 			break;
 		case DRM_FORMAT_XRGB1555:
 		case DRM_FORMAT_RGB565:
diff --git a/drivers/gpu/drm/vmwgfx/vmwgfx_kms.h b/drivers/gpu/drm/vmwgfx/vmwgfx_kms.h
index 3ee03227607c..3040e41ff3de 100644
--- a/drivers/gpu/drm/vmwgfx/vmwgfx_kms.h
+++ b/drivers/gpu/drm/vmwgfx/vmwgfx_kms.h
@@ -251,6 +251,8 @@ static const uint32_t vmw_primary_plane_formats[] = {
 	DRM_FORMAT_RGB888,
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
+	DRM_FORMAT_XBGR8888,
+	DRM_FORMAT_ABGR8888,
 };
 
 static const uint32_t vmw_cursor_plane_formats[] = {
-- 
2.17.1

